explorer/ast/static_scope.cpp

```diff
@@ -23,6 +23,12 @@ auto StaticScope::Add(std::string_view name, ValueNodeView entity,
     if (static_cast<int>(status) > static_cast<int>(it->second.status)) {
       it->second.status = status;
     }
+  } else {
+    if (trace_stream_->is_enabled()) {
+      *trace_stream_ << "--- declared `" << name << "` in `"
+                     << PrintAsID(entity.base()) << "` ("
```

Comment:
`entity` here is the entity that we're binding the name to, not the name of the scope, so "in `entity`" isn't really the right thing to be saying here.

I think it would be useful to add an `AstNode*` member to `StaticScope` to track the AST node whose scope we're referring to, so we can print that information out here. Then perhaps we could report something like:

```
--- declared `foo` as `fn N.Foo` in `namespace N` (foo.carbon:4)
```

Reply:
I'm sorry I missed that, thanks for mentioning it. 
I have one doubt, what kind of `AstNode*` should I use for **file scope**?